api/app/utils.py

```
@@ -0,0 +1,60 @@
+            aws_access_key_id=settings.r2_access_key_id,
+            aws_secret_access_key=settings.r2_secret_access_key,
+            region_name='auto',
+        )
+        s3.list_buckets()
+        return s3
+    except ClientError as e:
+        print(f"Error initializing S3 client: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Could not connect to R2 storage. Check credentials/configuration."
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)